core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -634,18 +634,43 @@ class ReassignPartitionsCommand(zkClient: KafkaZkClient,
         // Create reassignment znode so that controller will send LeaderAndIsrRequest to create replica in the broker
         zkClient.createPartitionReassignment(validPartitions.map({case (key, value) => (new TopicPartition(key.topic, key.partition), value)}).toMap)
 
-        // Send AlterReplicaLogDirsRequest again to make sure broker will start to move replica to the specified log directory.
-        // It may take some time for controller to create replica in the broker. Retry if the replica has not been created.
-        var remainingTimeMs = startTimeMs + timeoutMs - System.currentTimeMillis()
-        val replicasAssignedToFutureDir = mutable.Set.empty[TopicPartitionReplica]
-        while (remainingTimeMs > 0 && replicasAssignedToFutureDir.size < proposedReplicaAssignment.size) {
-          replicasAssignedToFutureDir ++= alterReplicaLogDirsIgnoreReplicaNotAvailable(
-            proposedReplicaAssignment.filter { case (replica, _) => !replicasAssignedToFutureDir.contains(replica) },
-            adminClientOpt.get, remainingTimeMs)
-          Thread.sleep(100)
-          remainingTimeMs = startTimeMs + timeoutMs - System.currentTimeMillis()
+        if (proposedReplicaAssignment.nonEmpty) {
```

Comment:
the tests in ReassignPartitionsCommandTest cover this PR. However, it is not stable since ReassignPartitionsCommand does not have correct "wait" for replica folder change.

Instead of checking return value of Admin#alterReplicaLogDirs, this PR get the replica folder via Admin#describeLogDirs and make sure the target folder is existent and isFuture = false